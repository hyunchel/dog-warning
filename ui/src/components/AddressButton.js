import React from 'react';
import { useHistory } from 'react-router-dom';

const AddressButton = (props) => {
    let history = useHistory();
    
    function handleDirect () {
        history.push("/address");
        console.log(history);
        props.handleSearch();
    }

    return(
        <button
            className={ props.className }
            onClick={ handleDirect }  
        >
        {/* <a href= { props.target }> */}
        { props.buttonValue }
        {/* </a> */}
        </button>
    );
}

export default AddressButton;