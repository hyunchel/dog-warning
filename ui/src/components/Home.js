import React from 'react';
import InputFormat from './formats/InputFormat';
import ButtonFormat from './formats/ButtonFormat';
import AddressButton from './formats/AddressButton';

const Home = (props) => {
    return(
        <div className="container">
            <div className="caution"><a>CAUTION</a> DOG WILL BITE</div>
            <div className="userInput">
            <InputFormat 
                className={ "addressInput "}
                placeholder={ "Please type address" }
                value={ props.address }
                name={ "address" }
                handleChange = { props.handleChange }
            />
            {/* <ButtonFormat 
                className={ "button search" }
                buttonValue={ "Search" }            
                handleSearch={ props.handleSearch }
                // target={ "/address" }
            /> */}
            <AddressButton  
                className={ "button search" }
                buttonValue={ "Search" }
                handleSearch={ props.handleSearch }
            />
            </div>
        </div>
    );
}

export default Home;