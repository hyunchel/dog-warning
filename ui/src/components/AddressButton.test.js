import React from 'react';
import AddressButton from './AddressButton';
import Home from './Home';
import renderer from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

it('renders correctly', () => {
    const history = createMemoryHistory();
    const tree = renderer
    .create(
        <Router history={history}>
            <Home>
                <AddressButton />
            </Home>
        </Router>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
})