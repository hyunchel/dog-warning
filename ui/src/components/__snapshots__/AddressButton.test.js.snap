// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="caution"
  >
    <a>
      CAUTION
    </a>
     DOG WILL BITE
  </div>
  <div
    className="userInput"
  >
    <input
      className="addressInput "
      name="address"
      placeholder="Please type address"
    />
    <button
      className="button search"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
